pkg/registry/resource/resourceclaim/strategy.go

```diff
@@ -220,3 +224,53 @@ func draAdminAccessFeatureInUse(claim *resource.ResourceClaim) bool {
 
 	return false
 }
+
+func dropDisabledDRAResourceClaimDeviceStatusFields(newClaim, oldClaim *resource.ResourceClaim) {
+	if !utilfeature.DefaultFeatureGate.Enabled(features.DRAResourceClaimDeviceStatus) {
+		newClaim.Status.Devices = nil
+	}
+}
+
+// dropDeallocatedStatusDevices removes the status.devices that were allocated
+// in the oldClaim and that have been removed in the newClaim.
+func dropDeallocatedStatusDevices(newClaim, oldClaim *resource.ResourceClaim) {
+	if !utilfeature.DefaultFeatureGate.Enabled(features.DRAResourceClaimDeviceStatus) {
+		return
+	}
```

Comment:
I think this also applies here 
```suggestion
isDRAResourceClaimDeviceStatusInUse := (oldClaim != nil && oldClaim.Status.Devices != nil)
	if !utilfeature.DefaultFeatureGate.Enabled(features.DRAResourceClaimDeviceStatus)  &&  !isDRAResourceClaimDeviceStatusInUse {
	return
```